yarn-project/txe/src/node/txe_node.ts

```diff
@@ -201,10 +201,21 @@ export class TXENode implements AztecNode {
     // hold a reference to them.
     // We should likely migrate this so that the trees are owned by the node.
 
-    // TODO: blockNumber is being passed as undefined, figure out why
+    if (blockNumber === undefined) {
+      throw new Error(
+        `Trying to get leaves indexes in the TXe with an undefined block number - this should never happen`,
+      );
+    }
+
+    if (blockNumber === (await this.getBlockNumber())) {
+      throw new Error(
+        'This is also weird, the node being requested for state that is being built (in the TXe, the blockNumber is the block currently being built (not committed yet)) - this should never happen.',
```

Comment:
Since these are very strange error I would try to make the messages more descriptive (specifically try not to use "also", since they won't happen simultanously)